### PR TITLE
Temp benchmark ci dir

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR adds a temporary dir for running benchmarks. Otherwise github action will remove the repo's temp dir, causing runs to fail like https://wandb.ai/costa-huang/cleanRL/runs/td84rput/logs. With this PR we can actually run more than one benchmark CI at a time. 

`FileNotFoundError: [Errno 2] No such file or directory: b'runs/CartPole-v1__ppo__1__1694627386/events.out.tfevents.1694627394.ip-26-0-147-6.516615.0'`

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -28,10 +28,13 @@ jobs:
...
-      - name: Checkout PR branch ${{ steps.comment-branch.outputs.head_ref }}
+      - name: Checkout `main` branch
         uses: actions/checkout@v3
-        with:
-          ref: ${{ steps.comment-branch.outputs.head_ref }}
+      - name: Checkout PR branch
+        run: gh pr checkout $PR_NUMBER
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.issue.number }}
```
Comment (younesbelkada) [inline .github/workflows/benchmark.yml:37]: These changes seem to be also introduced in #764

**Review by younesbelkada — APPROVED** [review]
Thanks!

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the results

**Comment by vwxyzjn** [comment]
**[COSTA BENCHMARK BOT]**: Here are the results

**Comment by vwxyzjn** [comment]
@younesbelkada, it looks like (almost) everything works as expected. I'm not entirely sure why the bot sent two images, but it feels good enough for now :)

**Pull request merged**